v3.0_roadmap.adoc

```diff
@@ -0,0 +1,243 @@
+= Road map for James 3.0
+
+This document aims at dividing existing and upcoming code into the following categories :
+
+ - supported
+ - experimental
+ - unsupported
+
+== Mailbox section
+
+.Feature matrix : mailbox backend
+|===
+|Feature name |Status | Comment
+
+|JPA
+|supported
+|Already existing code
+
+|Memory
+|Unsupported
+|Reference for testing
+
+|Cassandra
+|experimental
+|Essentially we do not know about how performance will behave. Also consistency issues might arise.
+
+|ElasticSearch
+|supported
+|
+
+|HBase
+|unsupported
+|
+
+|JCR
+|unsupported
+|
+
+|Lucene
+|unsupported
+|
+
+|Caching
+|unsupported
+|
+
+|Maildir
+|unsupported
+|Known issue on MailboxPath handling
+
+|===
+
+What being "supported implies for a backend ?
+
+ - Being MPT tested
+ - Deployement tests part of unit tests (ex : cassandra-guice)
+ - Having full support in server/data/data-@backend@
+ - Having its mappers passing generic unit tests for mappers
+ - It should at least support USER_FLAGS and NAMESPACE features on MPT tests
+ - Be able to handle a constant gatling workload for at least 4 hours with constant latency
+ - Be tested with thunderbird (we should write scenari for that)
+
+
+.Feature matrix : mailbox capabilities
+|===
+|Feature name |Status | Comment
+
+|ACL
+|unsupported
+|
+
+|Shared mailbox
+|unsupported
+|
+
+|Quota
+|experimental
+|
+
+|User flags
+|supported
+|
+
+|Distributed events
+|Unsupported
+|
+
+|===
+
+=== Short note about what needs to be DONE
+
+**General** :
+
+ - Define Thunderbird scenari
+ - Write Gatling tests for IMAP
+
+**Cassandra** :
+
+ - We need a Cassandra implementation for MailRepository
+
+**Memory** :
+
+ - We need a Memory SieveRepository
+ - We need a Memory implementation for MailRepository
+
+**JPA** :
+
+ - We need JAMES-1693-2 to be merged.
+ - We need JPA implementation for SieveRepository and MailRepository.
+ - We need JPA to valid MessageMapper and MailboxMapper unit tests.
+
+Bonus : have quota support with JPA...
+
+== Server
+
+.Injection framework
+|===
+|Name |Status
+
+|Spring
+|Supported
+
+|Guice cassandra
+|experimental
+
+|Guice memory
+|experimental
+
+|Cluster deployement
+|unsupported
+|===
+
+We need tests to ensure James starts when Spring is used.
+
+.Components
+|===
+|Name |Status |Comment
+
+|ActiveMQ mail queue
+|supported
+|
+
+|Camel mail processing
+|supported
+|
+
+|Sieve
+|experimental
+|Problem : it is compulsory today to rely on SIEVE scripts upon message reception
```

Comment:
If you tag SIEVE to supported, then you also need a supported way to manage your scripts. And ManageSieve is tagged as unsupported.

More stuff might not be 100% conform to RFCs. I have doubts on multi-line text. I know I missed the "notifications should not be sent twice" on vacation. My vacations replies do not have the Auto-Submitted auto-replied header.

We do not have load tests on this. 

We do not have to my knowledge any mecanism appart quotas to limit script complexity (and it is not quota job, and not its intent).